pkg/cosign/keys.go

```diff
@@ -235,3 +241,18 @@ func LoadPrivateKey(key []byte, pass []byte) (signature.SignerVerifier, error) {
 		return nil, errors.New("unsupported key type")
 	}
 }
+
+// given already decrypted blob with x509 encoded private key, try different x509.Parse<*>PrivateKey
+// functions to load it.
+func parsePrivateKey(key []byte) (crypto.PrivateKey, error) {
```

Comment:
Can we revert this change, since it should only be for PKCS8 encoded keys?

Reply:
rolled back in [858c9c6](https://github.com/sigstore/cosign/pull/3776/commits/858c9c6d037a2263d2d857c566dc0f71a3974aff)  (except comments) - kept the additional test to make clear which key formats are supported. Additionally created https://github.com/sigstore/cosign/issues/3779 for the task of moving functions to an internal package.